app/presenters/hyrax/permission_badge.rb

```diff
@@ -1,62 +1,38 @@
 module Hyrax
   class PermissionBadge
     include ActionView::Helpers::TagHelper
-    include HyraxHelper
 
-    def initialize(solr_document)
-      @solr_document = solr_document
+    # @param visibility_or_document [String,#visibility] the current visibility or an object
+    #                                                    that has a method returning visibility
+    def initialize(visibility_or_document)
+      @visibility = if visibility_or_document.respond_to?(:visibility)
```

Comment:
Extract if statement into a method; If someone were extend PermissionBadge it would provide a place to change.

(As a matter of practice, I prefer minimal business logic in initialize, instead preferring methods for any complex setter decisions)